### PR TITLE
Add new activities

### PR DESCRIPTION
**Pull request by mrjslau**
Added ShoppingCart and ShoppingList activities. Added custom preferences manager class. Changed file structure to get a cleaner look.

- ShoppingCart is persistent - list is saved to user's prefs file and all the data is available after the app is reopened.
- ShoppingList is not persistent yet.

Everything works as it should, tested on emulator and real life device.

++FIXED crazy gitignore line

Also found a bug - when reopening the app java error pops out. (Was there before my changes).

### CODE REVIEW

File: Mobile/ShopLens/Droid/Source/CameraActivity.cs [hunk 1]
```diff
@@ -57,6 +57,10 @@ protected override void OnCreate(Bundle savedInstanceState)
 
             SetContentView(Resource.Layout.Camera);
 
+            //var toolbar = FindViewById<Android.Support.V7.Widget.Toolbar>(Resource.Id.toolbar);
+            //SetSupportActionBar(toolbar);
+            //SupportActionBar.Title = "ShopLens";
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/Source/CameraActivity.cs]: Forgot to remove these comments probably?

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using Android.Content;
+
+namespace ShopLens.Droid.Source
+{
+    public class ActivityPreferences
+    {
+        ISharedPreferences prefs;
+        ISharedPreferencesEditor prefsEditor;
+        Context pContext;
+        List<string> items;
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/Source/AppPreferences.cs]: `items` might be just a local variable, it's not used anywhere except `GetPreferencesToList` I think

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        int itemCounter = 0;
+
+        public ActivityPreferences(Context context, string name)
+        {
+            pContext = context;
+            prefs = pContext.GetSharedPreferences(name, FileCreationMode.Private);
+            prefsEditor = prefs.Edit();
+            foreach (KeyValuePair<string, object> entry in prefs.All)
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/Source/AppPreferences.cs]: Can't open the IDE right now, but isn't there a simpler way to do this? Like `itemCounter = prefs.Count` or something like that?

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        public List<string> GetPreferencesToList()
+        {
+            if (items == null)
+            {
+                items = new List<string> { };
+            }
+            foreach (KeyValuePair<string, object> entry in prefs.All)
+            {
+                items.Add(entry.Value.ToString());
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/Source/AppPreferences.cs]: Maybe these lines (27 - 34) can be done with LINQ? Not that it's bad, but just as LINQ practice for the exam

File: Mobile/ShopLens/Droid/Source/ShoppingListActivity.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            listView.Adapter = listAdapter;
+            listView.ChoiceMode = ChoiceMode.Multiple;
+
+            addItemButton.Click += (sender, e) =>
+            {
+                if (!String.IsNullOrWhiteSpace(addItemEditText.Text))
+                {
+                    listAdapter.Add(addItemEditText.Text);
+                }
+                listAdapter.NotifyDataSetChanged();
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/Source/ShoppingListActivity.cs]: In here I imagine "NotifyDataSetChanged" updates the `listView` elements?

**Review by EdvardasDlugauskas — APPROVED** [review]
Looks quite good 👍 I liked that you found a way to store shopping items locally (I think?)

Maybe we should put the Activity files in separate folder, I think MJay did that when he was restructuring.

Also, I can't review with an IDE at the moment, so maybe @MellowJay or @s1712333 could have a look as well

**Review by marjasi — APPROVED** [review]
Everything looks good to me 👍. I say we merge!

**Comment by EdvardasDlugauskas** [comment]
Ok. @mrjslau or @MellowJay could you merge and then move activities .cs files to separate "Activities" folder? You can commit that straight to DEV (just make sure it builds!) 

Oh, and resolve the branch conflict

File: Mobile/ShopLens/Droid/Source/CameraActivity.cs (hunk 1 above)
Comment (mrjslau) [reply Mobile/ShopLens/Droid/Source/CameraActivity.cs]: Fixed.

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs (hunk 2 above)
Comment (mrjslau) [reply Mobile/ShopLens/Droid/Source/AppPreferences.cs]: Fixed.

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs (hunk 3 above)
Comment (mrjslau) [reply Mobile/ShopLens/Droid/Source/AppPreferences.cs]: Fixed.

File: Mobile/ShopLens/Droid/Source/AppPreferences.cs (hunk 4 above)
Comment (mrjslau) [reply Mobile/ShopLens/Droid/Source/AppPreferences.cs]: Noted for later.

File: Mobile/ShopLens/Droid/Source/ShoppingListActivity.cs (hunk 5 above)
Comment (mrjslau) [reply Mobile/ShopLens/Droid/Source/ShoppingListActivity.cs]: It notifies the adapter that it's data set has been modified and it should refresh the listview that the adapter fills with data.

**Comment by mrjslau** [comment]
[[Activities] Create activities representing missing Forms functionality](https://trello.com/c/2cEzXKne/67-activities-create-activities-representing-missing-forms-functionality)

**Pull request merged**